File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteEngineerVisit/QuoteEngineerVisitSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace QuoteEngineerVisits

        Public Class QuoteEngineerVisitSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"

            Public Function [Insert](ByVal qEngineerVisit As QuoteEngineerVisit) As QuoteEngineerVisit
                _database.ClearParameter()
                _database.AddParameter("@QuoteJobOfWorkID", qEngineerVisit.QuoteJobOfWorkID, True)
                _database.AddParameter("@StatusEnumID", qEngineerVisit.StatusEnumID, True)
                _database.AddParameter("@NotesToEngineer", qEngineerVisit.NotesToEngineer, True)

                qEngineerVisit.SetQuoteEngineerVisitID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("QuoteEngineerVisit_Insert"))
                qEngineerVisit.Exists = True
                Return qEngineerVisit
            End Function

            Public Function QuoteEngineerVisits_Get_For_QuoteJob_Of_Work(ByVal QuoteJobOfWorkID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@QuoteJobOfWorkID", QuoteJobOfWorkID, True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("QuoteEngineerVisits_Get_For_QuoteJob_Of_Work")
                dt.TableName = Entity.Sys.Enums.TableNames.tblQuoteEngineerVisit.ToString
                Return New DataView(dt)
            End Function

            Public Function QuoteEngineerVisits_Get_For_QuoteJob_Of_Work_As_Objects(ByVal QuoteJobOfWorkID As Integer) As ArrayList
                Dim engineerVisits As New ArrayList
                For Each row As DataRow In QuoteEngineerVisits_Get_For_QuoteJob_Of_Work(QuoteJobOfWorkID).Table.Rows
                    Dim visit As New QuoteEngineerVisits.QuoteEngineerVisit
                    visit.IgnoreExceptionsOnSetMethods = True
                    visit.Exists = True
                    visit.SetQuoteEngineerVisitID = row.Item("QuoteEngineerVisitID")
                    visit.SetQuoteJobOfWorkID = row.Item("QuoteJobOfWorkID")
                    visit.SetStatusEnumID = row.Item("StatusEnumID")
                    visit.SetNotesToEngineer = row.Item("NotesToEngineer")
                    visit.SetDeleted = row.Item("Deleted")

                    engineerVisits.Add(visit)
                Next

                Return engineerVisits
            End Function

#End Region

        End Class

    End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/Quotes/QuoteEngineerVisit/QuoteEngineerVisitSQL.cs
using System.Collections;
using System.Data;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace QuoteEngineerVisits
    {
        public class QuoteEngineerVisitSQL
        {
            private Sys.Database _database;

            public QuoteEngineerVisitSQL(Sys.Database database)
            {
                _database = database;
            }

            
            public QuoteEngineerVisit Insert(QuoteEngineerVisit qEngineerVisit)
            {
                _database.ClearParameter();
                _database.AddParameter("@QuoteJobOfWorkID", qEngineerVisit.QuoteJobOfWorkID, true);
                _database.AddParameter("@StatusEnumID", qEngineerVisit.StatusEnumID, true);
                _database.AddParameter("@NotesToEngineer", qEngineerVisit.NotesToEngineer, true);
                qEngineerVisit.SetQuoteEngineerVisitID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("QuoteEngineerVisit_Insert"));
                qEngineerVisit.Exists = true;
                return qEngineerVisit;
            }

            public DataView QuoteEngineerVisits_Get_For_QuoteJob_Of_Work(int QuoteJobOfWorkID)
            {
                _database.ClearParameter();
                _database.AddParameter("@QuoteJobOfWorkID", QuoteJobOfWorkID, true);
                var dt = _database.ExecuteSP_DataTable("QuoteEngineerVisits_Get_For_QuoteJob_Of_Work");
                dt.TableName = Sys.Enums.TableNames.tblQuoteEngineerVisit.ToString();
                return new DataView(dt);
            }

            public ArrayList QuoteEngineerVisits_Get_For_QuoteJob_Of_Work_As_Objects(int QuoteJobOfWorkID)
            {
                var engineerVisits = new ArrayList();
                foreach (DataRow row in QuoteEngineerVisits_Get_For_QuoteJob_Of_Work(QuoteJobOfWorkID).Table.Rows)
                {
                    var visit = new QuoteEngineerVisit();
                    visit.IgnoreExceptionsOnSetMethods = true;
                    visit.Exists = true;
                    visit.SetQuoteEngineerVisitID = row["QuoteEngineerVisitID"];
                    visit.SetQuoteJobOfWorkID = row["QuoteJobOfWorkID"];
                    visit.SetStatusEnumID = row["StatusEnumID"];
                    visit.SetNotesToEngineer = row["NotesToEngineer"];
                    visit.SetDeleted = Conversions.ToBoolean(row["Deleted"]);
                    engineerVisits.Add(visit);
                }

                return engineerVisits;
            }

            
        }
    }
}